server/src/lib/model/db/utils/getDiff.ts

```diff
@@ -0,0 +1,25 @@
+export type IdDifferences = {
+  idsToBeRemoved: number[];
+  idsToBeAdded: number[];
+};
+
+/**
+ * Get the difference of two arrays
+ */
+export const getDifferenceInIds = (
+  requestedIds: number[],
+  savedIds: number[]
+): IdDifferences => {
+  const idsToBeAdded: number[] = [];
+  const idsToBeRemoved: number[] = [];
+
+  requestedIds?.map(id => {
```

Comment:
```suggestion
  requestedIds.forEach(id => {
```

We don't need `?` cause we will always have an array? And I think a forEach makes more sense here as it isnt being used to create a new array